.github/workflows/pr.yaml

```diff
@@ -105,9 +161,10 @@ jobs:
       extra-repo-deploy-key: CUGRAPH_OPS_SSH_PRIVATE_DEPLOY_KEY
       node_type: cpu32
   wheel-tests-pylibcugraph:
-    needs: wheel-build-pylibcugraph
+    needs: [wheel-build-pylibcugraph, changed-files]
     secrets: inherit
     uses: rapidsai/shared-workflows/.github/workflows/wheels-test.yaml@branch-24.10
+    if: needs.changed-files.outputs.test_python == 'true'
```

Comment:
This change results in the `cugraph` wheel not being built if Python tests aren't needed, because `wheel-build-cugraph` depends on `wheel-tests-pylibcugraph`. Is it OK to skip building the `cugraph` wheel in that case?

Reply:
> Is it OK to skip building the cugraph wheel in that case?

I think it is ok, yes.

But only if I've understood how this works correctly.

Seeing these paths in the `test_python` config:

```text
CONTRIBUTING.md
README.md
docs/**
img/**
notebooks/
```

I'm assuming that translates to "if any file was changed that doesn't match those patterns, then run the builds and test".

I cannot think of a case where a PR touching only those files would require rebuilding wheels...given that the notebooks are tested with conda packages, not wheels.

Reply:
Yes, I think that is OK, although it is a bit incongruous to build pylibcugraph but not cugraph. We could also reorder the jobs to make `wheel-build-cugraph` depend on `wheel-build-pylibcugraph` instead.

Reply:
If nobody has any objections, I'll merge this as-is, and we can consider revisiting the build graph at some point in the future if it becomes a problem.